x-pack/plugins/security_solution/public/flyout/left/context.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { createContext, useContext } from 'react';
+import type { LeftPanelProps } from '.';
+
+export interface LeftPanelContext {
+  /**
+   * Id of the document
+   */
+  eventId: string;
+  /**
+   * Name of the index used in the parent's page
+   */
+  indexName: string;
+}
+
+export const LeftFlyoutContext = createContext<LeftPanelContext | undefined>(undefined);
+
+export type LeftPanelProviderProps = {
+  /**
+   * React components to render
+   */
+  children: React.ReactNode;
+} & Partial<LeftPanelProps['params']>;
+
+export const LeftPanelProvider = ({ id, indexName, children }: LeftPanelProviderProps) => {
+  const contextValue = {
```

Comment:
I would wrap these context objects in a `useMemo` as any time a parent re-renders a new object is going to be created, even if the values stay the same. Alternatively, you could wrap the entire provider in a `React.memo`

Reply:
Also, since the context can be undefined...maybe something like this to avoid the casting?

```
  const contextValue = useMemo(
    () =>
      id && indexName
        ? {
            eventId: id,
            indexName,
          }
        : undefined,
    [id, indexName]
  );
  ```

Reply:
yup I like this a lot better, thanks!